src/executor/thread_pool.cpp

```diff
@@ -0,0 +1,69 @@
+#include <condition_variable>
+#include <mutex>
+#include <yaclib/executor/executor.hpp>
+#include <yaclib/executor/thread_pool.hpp>
+
+#include "../container/intrusive_list.hpp"
+
+namespace yaclib::executor {
+
+namespace {
+
+class ThreadPool final : public IExecutor {
+ public:
+  ThreadPool(IThreadFactoryPtr factory, size_t min_threads_count,
+             size_t max_threads_count)
+      : _min_threads_count{min_threads_count},
+        _max_threads_count{max_threads_count} {
+    for (size_t i = 0; i != _min_threads_count; ++i) {
+      _threads.PushBack(factory
```

Comment:
- [ ] Fix format in this lines

Try using clang-format
If it doesn't work out in any way, turn off auto-formatting for these lines